tls/s2n_cipher_suites.c

```diff
@@ -131,6 +136,11 @@ static int s2n_set_cipher_as_server(struct s2n_connection *conn, uint8_t *wire,
                     continue;
                 }
 
+                /* Don't choose a suite the client shouldn't be able support */
+                if (conn->client_protocol_version < match->minimum_required_tls_version) {
+                    continue;
+                }
+
```

Comment:
We could store a "candidate" cipher in that if statement and select it if we fall out of the outer loop with no matches(right before S2N_ERR_CIPHER_NOT_SUPPORTED).

In practice, we don't need to worry about SSLv3 client hellos. So, this would only help if a TLSv1/TLSv1.1 client hello ONLY had TLSv1.2 ciphers in its list.